core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -880,6 +880,9 @@ static void loadFileMetadataInternal(RpcContext rpcContext, LockedInodePath inod
       }
       ufsLength = ((UfsFileStatus) context.getUfsStatus()).getContentLength();
       long blockSize = ((UfsFileStatus) context.getUfsStatus()).getBlockSize();
+      if (ServerConfiguration.getBoolean(PropertyKey.USER_BLOCK_SIZE_ENABLED)) {
```

Comment:
I think this only reads the server-side default, so it doesn't take your path config or configuration from your client command. This defeats the purpose of having this property.

@beinan @yuzhu Do you know how to make this respect both the client-side config and path config? I'm afraid making this respect user-side config needs this property value to be in `message FileSystemCommonPOptions`. Getting the path config is easier than modifying RPC, but I forgot how :(

Reply:
In order to pick up the correct path-specific setting, we will need to resolve the config for the path. The path configs are returned from master https://github.com/Alluxio/alluxio/blob/ef6b7efffb2e18246b7fae4b73551db73c8b7943/core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java#L430
to the client
https://github.com/Alluxio/alluxio/blob/ef6b7efffb2e18246b7fae4b73551db73c8b7943/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java#L433
Then the client will resolve the path config. This is how you set `sync.interval` using `alluxio pathConf add /path` then how that takes effect.

The annoying thing is, in the sync process the master is actually the client that needs to do the path resolution, and currently there's no API you can directly call. @maobaolong I will take over this part and add one API for you to use. So in your PR here it should be just an easy call.